core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -978,7 +979,7 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean) extends Abstra
       s"${KafkaConfig.AdvertisedListenersProp} protocols must be equal to or a subset of ${KafkaConfig.ListenersProp} protocols. " +
       s"Found ${advertisedListeners.keySet}. The valid options based on currently configured protocols are ${listeners.keySet}"
     )
-    require(interBrokerProtocolVersion.onOrAfter(ApiVersion(messageFormatVersion)),
+    require(interBrokerProtocolVersion >= messageFormatVersion,
       s"message.format.version $messageFormatVersion cannot be used when inter.broker.protocol.version is set to $interBrokerProtocolVersion")
```

Comment:
Currently, the message in the log4j log is a bit weird if the inter.broker.protocol/message.format is set to 0.10.0 since it will log 0.10.0-IV0. Perhaps we can just log the original string specified in the config.